lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/runner/Command.scala

```diff
@@ -27,6 +28,23 @@ case class Command(command: Seq[String], extraEnv: Seq[(String, String)]) {
       process.waitFor()
     }
 
+  /** Runs the command and returns its exit code along with any output produced..
+    *
+    * May return an exception if it is impossible to run the command (for
+    * example due to insufficient permissions or nonexistent executable).
+    */
+  def runAndCaptureOutput(): Try[(Int, String)] =
+    wrapError {
+      logger.debug("Executing {}", this)
+      val processBuilder = builder()
+      val process        = processBuilder.start()
+      val wrappedProcess = new WrappedProcess(command, process)
+      val stringBuilder  = new StringBuilder()
+      wrappedProcess.appendToBuilder(stringBuilder)
+      val result = wrappedProcess.join(waitForDescendants = true, 5 * 60)
```

Comment:
I'm not sure if that timeout is enough. The timeout was primarily needed in tests. But in the runner, depending on the operation the runtime can be very large:
1. a REPL can stay open for hours
2. a script can keep executing for a long time
3. a dependency preinstall can take a long time over weak connection / large deps.
All these should not have a timeout as their runtime is unbounded.

I guess only `new project` creation is something that is expected to be rather fast and could indeed be OK with a timeout.

I'd suggest to move the timeout parameter to be a parameter of `runAndCaptureOutput` so that we can only set it in places where we know it is OK. Plus, let's make it an `Option` so that one can set no timeout at all.
 As long as it is implicit like currently, it will get forgotten at some point.

Reply:
As I told above - there is many 'somethings' that may very well run for more than 5 minutes and that is not an error. Only short operations like new project are indeed small enough that running for this long is definitely an error.

That's why I wanted to make the distinction clear. Currently at call-site it is completely not clear that `run` has no time limit but `runAndCaptureOutput` does have one. That's why I think we should put the timeout in the signature to make that clearer. Or at least add a note in doc comment.

Reply:
To better explain my worry about this - looking at the API the only difference between `run` and `runAndCaptureOutput` seems to be the 'capturing stdout' aspect. But that is **not true** - they also differ by the fact that one of them offers unbounded execution time whereas the other will timeout. This is not clear from the API. Thus I'm worried that in the future someone may replace a `run` method call with `runAndCaptureOutput` and not realize that this also implies limiting the run-time of the underlying command. As we **do have** long running commands in there, this may be incorrect. And it is a kind of thing that tests will not discover as indeed we don't have any tests that check that the thing will run OK for a longer time.